atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -448,10 +448,11 @@ private static LockAndTimestampServices withRefreshingLockService(
 
     private static LockAndTimestampServices withRequestBatchingTimestampService(
             java.util.function.Supplier<TimestampClientConfig> timestampClientConfigSupplier,
-            LockAndTimestampServices lockAndTimestampServices) {
-        TimelockService timelockServiceWithBatching =
+            LockAndTimestampServices lockAndTimestampServices, String userAgent) {
+        TimelockService timelockServiceWithBatching = ServiceCreator.createInstrumentedService(
                 DecoratedTimelockServices.createTimelockServiceWithTimestampBatching(
-                        lockAndTimestampServices.timelock(), timestampClientConfigSupplier);
+                        lockAndTimestampServices.timelock(), timestampClientConfigSupplier),
+                TimelockService.class, userAgent);
```

Comment:
Agree we should have some testing for this. But, I think that this is actually resulting in double-wrapping. We create an instrumented feign proxy for the timelock service, and then we wrap that in a timestamp batching timelock service, and finally we wrap the whole thing in instrumentation again.

@hsaraogi I think we probably do want the instrumentation on the batching service as you've done here, but to avoid double-wrapping maybe we should actually instrument the `RequestBatchingTimestampService` (which is created in `DecoratedTimelockServices`)